controllers/capabilities/proxyconfigpromote_controller.go

```diff
@@ -176,30 +195,34 @@ func (r *ProxyConfigPromoteReconciler) proxyConfigPromoteReconciler(proxyConfigP
 			}
 			latestStagingVersion = stageElement.ProxyConfig.Version
 
+			// Retrieving latest production version
 			productionElement, err := threescaleAPIClient.GetLatestProxyConfig(productIDStr, "production")
 			if err != nil {
 				reqLogger.Info("Error while finding production version")
 				if !threescaleapi.IsNotFound(err) {
-					//status := r.proxyConfigPromoteStatus(productIDStr, proxyConfigPromote, err, reqLogger, "Failed")
 					statusReconciler := NewProxyConfigPromoteStatusReconciler(r.BaseReconciler, proxyConfigPromote, "Failed", productIDStr, 0, latestStagingVersion, err)
 					return statusReconciler, err
 				}
 			}
 			latestProductionVersion = productionElement.ProxyConfig.Version
 
+			// Promoting staging latest to production
 			_, err = threescaleAPIClient.PromoteProxyConfig(productIDStr, "sandbox", strconv.Itoa(stageElement.ProxyConfig.Version), "production")
 			if err != nil {
-				statusReconciler := NewProxyConfigPromoteStatusReconciler(r.BaseReconciler, proxyConfigPromote, "Failed", productIDStr, latestProductionVersion, latestStagingVersion, err)
+				// The version can already be in the production meaning that it can't be updated again, the proxyPromote is not going to be deleted by the operator but instead, will notify the user of the issue
+				statusReconciler := NewProxyConfigPromoteStatusReconciler(r.BaseReconciler, proxyConfigPromote, string(capabilitiesv1beta1.ProxyPromoteConfigFailedConditionType), productIDStr, latestProductionVersion, latestStagingVersion, err)
```

Comment:
hey @austincunningham can you think of any other potentiall errors we could here apart from the comms error? 
When there's nothing to promote the error returned in status isn't too informative IMO and customer might find it confusing on why it's failing. I was thinking of adding a custom error message here - something of "can't promote to production since there's no staged changes". WDYT?

Reply:
I like that yes. Will update

Reply:
```
"can't promote to production as no product changes detected, delete the proxyConfigPromote CR or introduce changes to stage env first to proceed"
```